### PR TITLE
Reactupdate npm updates

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Bumps this up to v2.7.1-r16 - a React 16 specific version of the Header.

All the changes here have been submitted in PR #93  which was merged to the `master` branch.

This PR updates npm packages to fix security issues and also to bring it up to par with the `master` branch. For the most part, npm packages in this PR are updated to the same version that is in `master`. npm packages that are specific to this branch are related to React 16 (enzyme adapter, react, etc). Some syntax is updated based on changes to `master` from the  #93 PR.

Testing in development mode locally works through `npm run start:dev` but testing the production build locally currently does not work. This is an ongoing issue in `master` as well.

I tested this in the Library Card app which uses the `reactupdate` branch.

### Security updates:
`reactupdate` branch
found 545 vulnerabilities (499 low, 15 moderate, 30 high, 1 critical)

`reactupdate-npm-updates` PR
found 447 low severity vulnerabilities

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,17 +113,10 @@ $ npm install
...
 We use Webpack to fire off a hot-reloading development server. This allows for continuous code changes without the need to refresh your browser.
 
-This starts the server at localhost:3000 defaulting to `NODE_ENV=development`.
+This starts the server at localhost:8080 defaulting to `NODE_ENV=development`.
 
 ```sh
-$ npm start
-```
-
-### Accessibility with React-a11y
```
Comment (EdwinGuzman) [inline README.md:122]: `react-a11y` is deprecated and should be replaced with `@axe-core/react` but that's not for this PR.

File: README.md [hunk 2]
```diff
@@ -173,10 +166,12 @@ be given a tab-index of −1. This allows the element to receive programmatic fo
 while being ignored during normal navigation flow.
 
 ### Production Mode
+TODO: The following needs to be updated with the Webpack update and will currently not run.
```
Comment (EdwinGuzman) [inline README.md:169]: Existing issue in `master`.

File: package.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+  "version": "2.7.1-r16",
   "description": "NYPL Header as a React component.",
-  "main": "dist/components/Header/Header.js",
+  "main": "dist/index.min.js",
   "scripts": {
-    "start": "NODE_ENV=development node server.js",
-    "babel-build": "./node_modules/.bin/babel src -d dist && cp -r src/styles/ dist/styles",
+    "start": "NODE_ENV=production node server.js",
+    "start:dev": "NODE_ENV=development webpack-dev-server --open",
     "dist": "NODE_ENV=production ./node_modules/.bin/webpack --config webpack.config.js && cp -r src/styles/ dist/styles",
```
Comment (EdwinGuzman) [inline package.json:9]: Replacing `babel-build` with `dist`. This removed a lot of unminified files in `/dist` with a single `index.min.js` file.

File: src/components/DonateButton/DonateButton.js [hunk 4]
```diff
@@ -14,7 +13,7 @@ const DonateButton = ({ id, className, target, label, gaLabel, style }) => (
     className={className}
     href={target}
     onClick={() => utils.trackHeader('Donate', gaLabel)}
-    style={_extend(style, defaultStyles)}
+    style={{...style, ...defaultStyles}}
```
Comment (EdwinGuzman) [inline src/components/DonateButton/DonateButton.js:16]: Some weird issue probably coming from underscore.

File: test/Header.test.js [hunk 5]
```diff
@@ -48,17 +49,17 @@ describe('Header', () => {
           }
         })
         .catch((response) => {
-          if (response instanceof Error) {
-            console.warn(response.message);
+          if (response.response instanceof Error) {
```
Comment (EdwinGuzman) [inline test/Header.test.js:52]: Updated `axios` error response.

**Review by nypl-wluisi — APPROVED** [review]
I tested pulling in this branch on Location Finder app, and everything looks fine. I also pulled in this repo/branch and tested that as well, running `npm run start:dev` seems to work fine.

**Review by siberry — APPROVED** [review]
Looks good to me. Thanks, Edwin.

**Comment by siberry** [comment]
Just ran the test suite with this version and [this test](https://github.com/NYPL/discovery-front-end/blob/development/test/unit/Application.test.js#L42) is not passing anymore. The page looks fine. Anything jumping out to you?

**Comment by EdwinGuzman** [comment]
I don't know exactly why it's not working but it looks like in the test it's not rendering as `Header` but rather as a shorthand variable, in this case `s`:

Screen Shot 2021-02-01 at 10 01 32 AM

But, yes, it renders and works fine because under the hood, the name doesn't matter when it's rendering as it does for the test. You can update the test to the following to make it work:

`expect(component.find('#nyplHeader')).to.have.length(1);`

**Comment by EdwinGuzman** [comment]
If that change is okay with you, I'll merge. Otherwise, I'll keep it open to see if I can update its name when rendering.

**Comment by siberry** [comment]
> If that change is okay with you, I'll merge. Otherwise, I'll keep it open to see if I can update its name when rendering.

It's perfect! Feel free to merge

**Pull request merged**